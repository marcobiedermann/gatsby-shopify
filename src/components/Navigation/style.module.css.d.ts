declare const styles: {
  readonly navigation: string;
  readonly navigation__link: string;
  readonly navigation__list: string;
  readonly 'navigation__listInline': string;
};
export = styles;
