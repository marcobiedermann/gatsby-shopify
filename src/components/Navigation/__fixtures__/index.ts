import { NavigationProps } from '..';

const navigation: NavigationProps = {
  routes: [
    {
      name: 'Home',
      path: '/',
    },
  ],
};

export default navigation;
