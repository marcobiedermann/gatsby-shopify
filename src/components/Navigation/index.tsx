import clsx from 'clsx';
import { Link } from 'gatsby';
import React from 'react';
import { Route } from '../../constants/routes';
import {
  navigation,
  navigation__link,
  navigation__list,
  navigation__listInline,
} from './style.module.css';

export interface NavigationProps {
  inline?: boolean;
  routes: Route[];
}

function Navigation(props: NavigationProps) {
  const { inline, routes } = props;

  return (
    <nav className={navigation}>
      <ul
        className={clsx(navigation__list, {
          [navigation__listInline]: inline,
        })}
      >
        {routes.map((route) => (
          <li key={route.path}>
            <Link className={navigation__link} to={route.path}>
              {route.name}
            </Link>
          </li>
        ))}
      </ul>
    </nav>
  );
}

export default Navigation;
