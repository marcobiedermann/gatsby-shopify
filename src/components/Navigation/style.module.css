.navigation {
}

.navigation__link {
  text-decoration: none;
}

.navigation__list {
  list-style: none;
  margin: 0;
  padding: 0;
}

.navigation__list--inline {
  display: flex;
  gap: 1em 2em;
}
