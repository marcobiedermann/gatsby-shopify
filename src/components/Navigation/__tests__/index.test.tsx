import { render } from '@testing-library/react';
import React from 'react';
import Navigation from '..';
import navigation from '../__fixtures__';

describe('navigation component', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Navigation {...navigation} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
