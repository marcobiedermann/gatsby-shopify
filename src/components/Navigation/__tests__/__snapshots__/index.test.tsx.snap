// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navigation component should render correctly 1`] = `
<DocumentFragment>
  <nav
    class="navigation"
  >
    <ul
      class="navigation__list"
    >
      <li>
        <a
          class="navigation__link"
          href="/"
        >
          Home
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
